### DISCUSSION TITLE
Help with display - ValueError: brightness (0) must be between 1.0 and 10.0

### DISCUSSION THREAD

**Discussion by thewade** [make-all/tuya-local#2168] [Q&A]
I could use some help with a device display for plantsio_ivy_smart_planter.yaml.

The dps accepts values 0 to 100, where 0 means the display is off.

The entry from the yaml file:

```  
  - entity: light
    translation_key: display
    category: config
    dps:
      - id: 101
        type: integer
        name: brightness
        range:
          min: 0
          max: 100
```

When I try to turn the device off I get an error:

```
2024-07-30 21:05:19.281 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [546576435088] Unexpected exception
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 241, in handle_call_service
    response = await hass.services.async_call(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2731, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2774, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 999, in entity_service_call
    single_response = await _handle_entity_call(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1071, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/light/__init__.py", line 639, in async_handle_light_off_service
    await light.async_turn_off(**filter_turn_off_params(light, params))
  File "/config/custom_components/tuya_local/light.py", line 450, in async_turn_off
    await self._brightness_dps.async_set_value(self._device, 0)
  File "/config/custom_components/tuya_local/helpers/device_config.py", line 473, in async_set_value
    settings = self.get_values_to_set(device, value)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/tuya_local/helpers/device_config.py", line 944, in get_values_to_set
    raise ValueError(f"{self.name} ({value}) must be between {mn} and {mx}")
ValueError: brightness (0) must be between 1.0 and 10.0
```

Is this a mistake the yaml file that can be corrected? Or a bug?

A seconday complication is if auto brightness is set, then the brightness value is ignored, although it retains its value.

Therefor to properly turn off the display you must set auto brightness to off and brightness to zero.

Is it possible to set brightness to invalid or hidden when auto brightness is set?

**Comment by make-all** [comment]
I don't know where that error is coming from. The error message implies a value that is scaled by 10 and with a range 10-100, but the config file clearly has no scaling and a range 0-100 for that value. Perhaps showing the diagnostics for that device would shed some light.

**Reply by thewade** [reply]
I have updated to tuya-local 2024.10.0 and home assistant 2024.10.3 I am no longer seeing the error in the logs.

However the behavior seems the be same. If I try to toggle to the off position, it bonces back to the on position in home assistant. If I look at the brightness value in the slider it is 1%. There is no way to set it either off, or 0%.

The dps values in the diagnostic files however show it was 0. In the same diagnostic file the home assistant brightness value shows as 3 and state as on. If I set it to 100% then in the diagnostic file the dps value is 100, and the home assistant value is 255.

[tuya_local-01J34JHQWR7H35HR2C9ZYGECC3-Smart Planter-0%.json](https://github.com/user-attachments/files/17469194/tuya_local-01J34JHQWR7H35HR2C9ZYGECC3-Smart.Planter-0.json)
[tuya_local-01J34JHQWR7H35HR2C9ZYGECC3-Smart Planter-100%.json](https://github.com/user-attachments/files/17469207/tuya_local-01J34JHQWR7H35HR2C9ZYGECC3-Smart.Planter-100.json)

**Comment by make-all** [comment]
Are you trying to turn off a group of lights here?  It seems like most likely one of the other lights is complaining about the attempt to set brightness to zero.

**Reply by thewade** [reply]
It is an LCD display on a smart planter. I am only trying to toggle its display.

When I first created the PR for the device, I just had it as a number entity and it worked as expected having a slider with 0 turning the display off and 1-100 adjusting the brightness.

However, as a light entity I just cannot get it to work as expected. In older builds I was getting the value error. In recent builds the value error is gone and now I have the behavior from my October 21 comment.

The recent behavior is a dps value of 0 is getting mapped to a home assistant value of 3 and the device state of on. It should instead map to a state of off. In the off state I don’t know if it passes a brightness although I know home assistant only accepts 1 to 255 as brightness so I cannot map it as zero.

I can see this when I click download diagnostics:

```
    {
        "disabled": false,
        "disabled_by": null,
        "entity_category": "config",
        "device_class": null,
        "original_device_class": null,
        "icon": null,
        "unit_of_measurement": null,
        "state": {
        "entity_id": "light.smart_planter_display",
        "state": "on",
        "attributes": {
            "supported_color_modes": [
            "brightness"
            ],
            "color_mode": "brightness",
            "brightness": 3,
            "friendly_name": "Smart Planter Display",
            "supported_features": 0
        },
        "last_changed": "2024-11-24T16:53:47.162854+00:00",
        "last_reported": "2024-11-24T16:53:54.775975+00:00",
        "last_updated": "2024-11-24T16:53:54.660719+00:00"
        }
```

Where the dps is actuallly zero
```
    "cached_state": {
      "updated_at": 1732468386.616958,
      "101": 0,
      
```

Keeping this discussion open after the behavior changed probably just leads to confusion. I am pretty sure behavior is a bug and if you want me to open an issue and close this thread, I would be happy to.

**Reply by make-all** [reply]
It would be better as a bug I think, but it will need debug logs capturing the attempt to turn off the light, so I can see the service details and the steps the integration is going through, along with what it is eventually sending to find why it is ending up with a value of 3.

**Comment by thewade** [comment]
Created bug https://github.com/make-all/tuya-local/issues/2589
